extensions/binding/org.eclipse.smarthome.binding.homematic/src/main/java/org/eclipse/smarthome/binding/homematic/internal/communicator/server/BinRpcResponseHandler.java

```diff
@@ -45,23 +47,33 @@ public BinRpcResponseHandler(Socket socket, RpcResponseHandler<byte[]> rpcRespon
     @Override
     public void run() {
         try {
-            BinRpcMessage message = new BinRpcMessage(socket.getInputStream(), true, config.getEncoding());
-            logger.trace("Event BinRpcMessage: {}", message);
-            byte[] returnValue = rpcResponseHandler.handleMethodCall(message.getMethodName(),
-                    message.getResponseData());
-            if (returnValue != null) {
-                socket.getOutputStream().write(returnValue);
-            }
+            boolean isMaxAliveReached;
+            do {
+                BinRpcMessage message = new BinRpcMessage(socket.getInputStream(), true, config.getEncoding());
+                logger.trace("Event BinRpcMessage: {}", message);
+                byte[] returnValue = rpcResponseHandler.handleMethodCall(message.getMethodName(),
+                        message.getResponseData());
+                if (returnValue != null) {
+                    socket.getOutputStream().write(returnValue);
+                }
+                isMaxAliveReached = System.currentTimeMillis() - created > (config.getSocketMaxAlive() * 1000);
+            } while (!isMaxAliveReached);
+
         } catch (EOFException eof) {
-            // ignore
+            closeSocket();
         } catch (Exception e) {
             logger.error("{}", e.getMessage(), e);
```

Comment:
Indeed, ERROR logging is reserved for the framework. Please use WARNING in case the user is able to correct the situation and/or should be informed. Use DEBUG in all other cases.